onset_maproom/calc.py

```diff
@@ -33,6 +35,123 @@ def onset_date(daily_rain, early_start_day, early_start_month, search_days, rain
 #  onset_date = onset_date[time_coord] + onset_date  
   return onset_date
 
+# Time functions
+
+def daily_tobegroupedby_season(daily_data, start_day, start_month, end_day, end_month, time_coord="T"):
+  """Returns dataset ready to be grouped by with:
+  the daily data where all days not in season of interest are dropped
+  season_starts: an array where the non-dropped days are indexed by the first day of their season -- to use to groupby
+  seasons_ends: an array with the dates of the end of the seasons
+  Can then apply groupby on daily_data against seasons_starts, and preserving seasons_ends for the record
+  If starting day-month is 29-Feb, uses 1-Mar.
+  If ending day-month is 29-Feb, uses 1-Mar and triggers the option to open the right edge of the Interval.
+  That means that the last day included in the season will be 29-Feb in leap years and 28-Feb otherwise
+  """
+  #Deal with leap year cases
+  if start_day == 29 and start_month == 2 :
+    start_day = 1
+    start_month = 3
+  end_day2 = end_day
+  end_month2 = end_month
+  if end_day == 29 and end_month == 2 :
+    end_day2 = 1
+    end_month2 = 3
+  #Drop date outside very first and very last edges -- this ensures we get complete seasons with regards to edges, later on
+  start_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==start_day) & (daily_data[time_coord].dt.month==start_month)),
+    drop=True
+  )
+  end_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==end_day2) & (daily_data[time_coord].dt.month==end_month2)),
+    drop=True
+  )
+  daily_data = daily_data.sel(**{time_coord: slice(start_edges[0],end_edges[-1])})
+  start_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==start_day) & (daily_data[time_coord].dt.month==start_month)),
+    drop=True
+  )
+  end_edges = daily_data[time_coord].where(
+    ((daily_data[time_coord].dt.day==end_day2) & (daily_data[time_coord].dt.month==end_month2)),
+    drop=True
+  )
+  #Creates array of edges of the season that will form the bins
+  seasons_edges = xr.concat([start_edges, end_edges], "T_out", join="override")
```

Comment:
That's because I need the start and end of each season "paired" against a common dimension. start/end_edges are on different dimensions that are their very values.

Reply:
Not positive I understand what you're saying, but what you want is to give `end_edges` the same time coordinate as `start_edges`, it seems simpler/clearer to do something like `end_edges.assign_coords(T=start_edges["T"])`, rather than combining them with `concat` only to split them apart again.

Reply:
done